op-node/rollup/driver/state.go

```diff
@@ -442,3 +458,28 @@ type hashAndErrorChannel struct {
 	hash common.Hash
 	err  chan error
 }
+
+// checkForGapInUnsafeQueue checks if there is a gap in the unsafe queue and attempts to retrieve the missing payloads from the backup RPC.
+// WARNING: The sync client's attempt to retrieve the missing payloads is not guaranteed to succeed, and it will fail silently (besides
+// emitting warning logs) if the requests fail.
+func (s *Driver) checkForGapInUnsafeQueue(ctx context.Context) {
+	start, end := s.derivation.GetUnsafeQueueGap()
+	size := end - start
+
+	// If there is a gap in the queue and a backup sync client is configured, attempt to retrieve the missing payloads from the backup RPC
+	if size > 0 {
+		s.log.Warn("Unsafe queue gap detected", "start", start, "end", end, "size", size)
+		if s.L2SyncCl != nil {
+			s.log.Info("Attempting to fetch missing payloads from backup RPC", "start", start, "end", end, "size", size)
+
+			// Attempt to fetch the missing payloads from the backup unsafe sync RPC concurrently.
+			// Concurrent requests are safe here due to the engine queue being a priority queue.
+			// TODO: Should enforce a max gap size to prevent spamming the backup RPC or being rate limited.
+			for blockNumber := start; blockNumber < end; blockNumber++ {
+				s.L2SyncCl.FetchUnsafeBlock <- blockNumber
+			}
+		} else {
+			s.log.Warn("No backup unsafe sync RPC configured, cannot fetch missing payloads!")
```

Comment:
This log will trigger every 60s when running in normal mode, perhaps we only call this function if the client is not nil and sanity check in here that its not nil to prevent issues with refactoring?

Reply:
Actually nevermind, its behind the size check